File: packages/sb-uhq/src/self/func/message_edit.js
"use strict";

/**
 * Édite un message du selfbot. Si l'édition échoue (ex: DM), envoie un nouveau message.
 * @param {import("discord.js-selfbot-v13").Message} message
 * @param {string} content
 */
async function messageEdit(message, content) {
  try {
    return await message.edit(content);
  } catch {
    return await message.channel.send(content).catch(() => null);
  }
}

module.exports = { messageEdit };

File: packages/sb-uhq/app/func/message_edit.py
"""Édite un message du selfbot ; retombe sur un envoi si l'édition échoue (DM, etc.)."""

from __future__ import annotations


async def message_edit(message, content: str):
    try:
        return await message.edit(content=content)
    except Exception:
        try:
            return await message.channel.send(content)
        except Exception:
            return None
